connect/runtime/src/test/java/org/apache/kafka/connect/runtime/AbstractHerderTest.java

```diff
@@ -474,6 +476,25 @@ public void testReverseTransformConfigs() {
         assertFalse(reverseTransformed.get(0).containsKey(TEST_KEY3));
     }
 
+    @Test
+    public void testConfigProviderRegex() {
+        testConfigProviderRegex("${file:/tmp/somefile.txt:somevar}");
+        testConfigProviderRegex("\"${file:/tmp/somefile.txt:somevar}\"");
+        testConfigProviderRegex("plain.PlainLoginModule required username=\"${file:/tmp/somefile.txt:somevar}\"");
+        testConfigProviderRegex("plain.PlainLoginModule required username=${file:/tmp/somefile.txt:somevar}");
```

Comment:
How about adding some tests where multiple variables are used in a single value? I think the logic cares only if one can be found, but it still would be nice to verify.

Also, currently all of the variables are at the end -- how about some where they are at the beginning and middle with additional content besides the variable?

Also, what about other variable formats? Variables are of the form `${file:path:key}` (per [KIP-297](https://cwiki.apache.org/confluence/display/KAFKA/KIP-297%3A+Externalizing+Secrets+for+Connect+Configurations)), so maybe some without a path and very simple file, path, and key values.

Reply:
Added multiple cases here. Hope this helps. Thanks!

Reply:
Excellent!